<?xml version="1.0"?>
<launch>
  <arg name="script" default="nav_cloning_node_pytorch.py"/>
  <arg name="mode" default="change_dataset_balance"/>
  <arg name="initial_pose_x" default="-10.1856145815951" />
  <arg name="initial_pose_y" default="28.441895426996574" />
  <arg name="initial_pose_a" default="-1.57" />
  <arg name="robot_x" default="-10.1856145815951" />
  <arg name="robot_y" default="28.441895426996574" />
  <arg name="robot_Y" default="-1.57" />
  <arg name="use_waypoint_nav" default="true" />
  <rosparam param="/nav_cloning_node/mode" subst_value="True">$(arg mode)</rosparam>

  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find nav_cloning)/world/Tsudanuma_2-3.world"/> 
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
  </include>

  <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_waffle_pi.urdf.xacro" />

  <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf" args="-urdf -model mobile_base -x $(arg robot_x) -y $(arg robot_y) -z 0.0 -Y $(arg robot_Y) -param robot_description" />

  <node pkg="timed_roslaunch" type="timed_roslaunch.sh"
        args="5 nav_cloning nav_cloning.launch script:=$(arg script)"
        name="timed_roslaunch" output="screen" />

  <node pkg="timed_roslaunch" type="timed_roslaunch.sh"
        args="8 nav_cloning turtlebot3_navigation.launch model:=waffle map_file:=$(find nav_cloning)/maps/cit_3f_map.yaml waypoints_file:=$(find nav_cloning)/maps/cit3f_way.yaml initial_pose_x:=$(arg initial_pose_x) initial_pose_y:=$(arg initial_pose_y) initial_pose_a:=$(arg initial_pose_a) use_waypoint_nav:=$(arg use_waypoint_nav)"
        name="timed_roslaunch2" >
  </node>
  <!-- <node pkg="timed_roslaunch" type="timed_roslaunch.sh" -->
  <!--       args="8 nav_cloning turtlebot3_navigation.launch model:=waffle map_file:=$(find nav_cloning)/maps/willowgarage.yaml waypoints_file:=$(find nav_cloning)/maps/test_loop.yaml" -->
  <!--       name="timed_roslaunch2" > -->
  <!-- </node> -->

  <node pkg="timed_roslaunch" type="timed_roslaunch.sh"
        args="20 nav_cloning start_wp_nav.launch"
        name="timed_roslaunch3" >
  </node>
</launch>
